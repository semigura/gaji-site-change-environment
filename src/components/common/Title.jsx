import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import css from './Title.module.scss';

const Title = props => {
  const { className, children, tag, withColor } = props;
  const classes = classNames(css.Title, className);
  const titleClasses = classNames(
    css.Title__Text,
    withColor ? css.withColor : '',
  );
  const Tag = tag || 'h2';

  return (
    <div className={classes}>
      <Tag className={titleClasses}>{children}</Tag>
    </div>
  );
};

Title.propTypes = {
  className: PropTypes.string,
  children: PropTypes.string.isRequired,
  tag: PropTypes.oneOf(['h1', 'h3', 'h4', 'h5']),
  withColor: PropTypes.bool,
};

Title.defaultProps = {
  className: undefined,
  tag: undefined,
  withColor: false,
};

export default Title;
