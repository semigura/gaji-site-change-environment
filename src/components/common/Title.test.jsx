import { shallow } from 'enzyme';
import React from 'react';

import Title from './Title';

describe('<Title />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Title>Title</Title>);
    expect(wrapper).toMatchSnapshot();
  });

  describe('withColor', () => {
    it('Class name `withColor` is applied', () => {
      const wrapper = shallow(<Title withColor>Title</Title>);
      expect(wrapper.find('.withColor')).toHaveLength(1);
    });
  });

  describe('tag=h4', () => {
    it('Use h4 tag', () => {
      const wrapper = shallow(<Title tag="h4">Title</Title>);
      expect(wrapper.find('h4')).toHaveLength(1);
    });
  });
});
