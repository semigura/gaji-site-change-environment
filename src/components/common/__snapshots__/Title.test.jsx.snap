// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> renders correctly 1`] = `
<div
  className="Title"
>
  <h2
    className="Title__Text"
  >
    Title
  </h2>
</div>
`;
